uui-components/src/navigation/MainMenu/MainMenu.tsx

```diff
@@ -50,161 +56,123 @@ export interface MainMenuProps extends IHasCX, IHasRawProps<React.HTMLAttributes
-                        children={ hiddenItems?.filter((i) => i.props.collapseToMore).map((i) => i.render(item, hiddenItems)) }
-                    />
-                ),
-                collapsedContainer: true,
-            });
-        }
-
-        return items;
-    }
-
-    renderServerBadge() {
+export function MainMenu(props: MainMenuProps) {
+    const {
+        items,
+        appLogoUrl,
+        logoLink,
+        logoHref,
+        onLogoClick,
+        renderBurger,
+        alwaysShowBurger,
+        serverBadge,
+        Burger,
+        cx: cxProp,
+        rawProps,
+        forwardedRef,
+    } = props;
+
+    // Server badge rendering logic
+    const renderServerBadge = useCallback(() => {
+        if (!serverBadge) return null;
         let serverBadgeColor;
-        if (!this.props.serverBadge) {
-            return;
-        } else {
-            switch (this.props.serverBadge) {
-                case 'Dev':
-                    serverBadgeColor = '#39b7ac';
-                    break;
-                case 'QA':
-                    serverBadgeColor = '#a3c644';
-                    break;
-                case 'UAT':
-                    serverBadgeColor = '#937ebd';
-                    break;
-                default:
-                    serverBadgeColor = '#30b6dd';
-            }
+        switch (serverBadge) {
+            case 'Dev':
+                serverBadgeColor = '#39b7ac';
+                break;
+            case 'QA':
+                serverBadgeColor = '#a3c644';
+                break;
+            case 'UAT':
+                serverBadgeColor = '#937ebd';
+                break;
+            default:
+                serverBadgeColor = '#30b6dd';
         }
-
-        return this.props.serverBadge !== 'Prod' && this.props.serverBadge !== 'Public' && this.props.serverBadge !== 'Demo' ? (
-            <div className={ cx(uuiMainMenu.serverBadge) }>
-                <div className={ cx(uuiMainMenu.serverBadgeLabel) } style={ { background: serverBadgeColor } }>
-                    {this.props.serverBadge}
+        return serverBadge !== 'Prod'
+            && serverBadge !== 'Public'
+            && serverBadge !== 'Demo' ? (
+                <div className={ cx(uuiMainMenu.serverBadge) }>
+                    <div
+                        className={ cx(uuiMainMenu.serverBadgeLabel) }
+                        style={ { background: serverBadgeColor } }
+                    >
+                        {serverBadge}
+                    </div>
                 </div>
-            </div>
-        ) : null;
-    }
+            ) : null;
+    }, [serverBadge]);
 
-    getMenuItems(): AdaptiveItemProps[] {
-        const Burger = this.props.Burger;
-
-        const items: AdaptiveItemProps[] = this.convertReactChildrenToItems(this.props.children) || [];
-
-        if (this.props.appLogoUrl) {
-            items.unshift({
+    const getMenuItems = useCallback((): AdaptiveItemProps[] => {
```

Comment:
need to think do we need getMenuItems? In current implementation is almost useless since it used only if props.items isn't provided, what is very rare case.